c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -180,6 +185,43 @@ struct AllocParams {
 
 } // namespace
 
+
+class CachingAllocatorConfig {
+public:
+  size_t max_split_size;  // largest "large" alloc, "oversize" is not splitable
+
+  CachingAllocatorConfig()
+  : max_split_size(SIZE_MAX)
```

Comment:
can you please use std::numeric_limits to not rely on ambiguous macro definitions?